IronFrame/ContainerHostService.cs

```diff
@@ -76,18 +76,18 @@ public IContainerHostClient StartContainerHost(string containerId, IContainerDir
                 Credentials = credentials,
             };
 
-            var hostProcess = processRunner.Run(hostRunSpec);
-
-            WaitForProcessToStart(hostProcess, HostProcessStartTimeout);
-
             // Order here is important.
-            // - Start the process and verify that it's running
+            // - Start the process 
             // - Add the process to the job object
+            // - Verify that the process is healthy
             // - Start the RPC message pump
             //
             // We need to ensure that the host process cannot create any new processes before
             // it's added to the job object.
+
+            var hostProcess = processRunner.Run(hostRunSpec);
             jobObject.AssignProcessToJob(hostProcess.Handle);
```

Comment:
Is there any chance for a race condition here, before the process itself has started but after a handle on the process is returned? I don't entirely understand the original reasoning behind the "important" order in the comment above, where the process was verified as running before adding it to the job object.